resources/assets/js/components/EventDevice.vue

```diff
@@ -270,6 +273,12 @@ export default {
         this.currentDevice.repair_details = null
       }
     },
+    partsProvider() {
+      // Third part parts are indicated via the parts provider field.
+      if (this.currentDevice.spare_parts === SPARE_PARTS_MANUFACTURER && this.currentDevice.parts_provider === PARTS_PROVIDER_THIRD_PARTY) {
```

Comment:
I think you're right.  Fix pushed and tested using that useful spec.

I've added something to the retro for fix.

Reply:
Still not working for me on display. The persisting is working fine, the correct values are stored, but when redisplayed it is always showing from manufacturer.  I think the logic for mapping back from spare parts / parts provider combo in to one of the three dropdown values is not right.

I think this would do it in the Vue component:
```
if (this.currentDevice.spare_parts === SPARE_PARTS_NOT_NEEDED) {
    this.currentDevice.spare_parts = SPARE_PARTS_NOT_NEEDED
} else {
  if (this.currentDevice.parts_provider === PARTS_PROVIDER_THIRD_PARTY) {
    this.currentDevice.spare_parts = SPARE_PARTS_THIRD_PARTY
  } else {
    this.currentDevice.spare_parts = SPARE_PARTS_MANUFACTURER
  }
}
```

Just for fun and to refresh my memory, I've added some tests around the creation of devices in the backend.

Reply:
Ok, next version pushed.

I agree this is horrible, and my brain refuses to get a grip on it, sorry.